internal/stanza/frompdataconverter.go

```diff
@@ -0,0 +1,273 @@
+	c.logger.Debug("Starting log converter from pdata", zap.Int("worker_count", cap(c.workerChan)))
+
+	for i := 0; i < cap(c.workerChan); i++ {
+		c.wg.Add(1)
+		go c.workerLoop()
+	}
+}
+
+func (c *FromPdataConverter) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopChan)
+		c.wg.Wait()
+		close(c.entriesChan)
+		close(c.workerChan)
+	})
+}
+
+// OutChannel returns the channel on which converted entries will be sent to.
+func (c *FromPdataConverter) OutChannel() <-chan []*entry.Entry {
+	return c.entriesChan
+}
+
+type fromConverterWorkerItem struct {
+	Resource       pdata.Resource
+	LogRecordSlice pdata.LogRecordSlice
+}
+
+// workerLoop is responsible for obtaining pdata logs from Batch() calls,
+// converting them to []*entry.Entry and sending them out
+func (c *FromPdataConverter) workerLoop() {
+	defer c.wg.Done()
+
+	for {
+		select {
+		case <-c.stopChan:
+			return
+
+		case workerItem, ok := <-c.workerChan:
+			if !ok {
+				return
+			}
+
+			select {
+			case c.entriesChan <- convertFromLogs(workerItem):
+			case <-c.stopChan:
+				return
+			}
+		}
+	}
+}
+
+// Batch takes in an set of pdata.Logs and sends it to an available worker for processing.
+func (c *FromPdataConverter) Batch(pLogs pdata.Logs) error {
+	for i := 0; i < pLogs.ResourceLogs().Len(); i++ {
+		rls := pLogs.ResourceLogs().At(i)
+		for j := 0; j < rls.ScopeLogs().Len(); j++ {
+			scope := rls.ScopeLogs().At(j)
+			item := fromConverterWorkerItem{
+				Resource:       rls.Resource(),
+				LogRecordSlice: scope.LogRecords(),
+			}
+			select {
+			case c.workerChan <- item:
+				continue
+			case <-c.stopChan:
+				return nil
+			}
+		}
+	}
+
+	return nil
+}
+
+// convertFromLogs converts the contents of a fromConverterWorkerItem into a slice of entry.Entry
+func convertFromLogs(workerItem fromConverterWorkerItem) []*entry.Entry {
+	result := make([]*entry.Entry, 0, workerItem.LogRecordSlice.Len())
+	for i := 0; i < workerItem.LogRecordSlice.Len(); i++ {
+		record := workerItem.LogRecordSlice.At(i)
+		entry := entry.Entry{}
+
+		entry.Resource = valueToMap(workerItem.Resource.Attributes())
+		convertFrom(record, &entry)
+		result = append(result, &entry)
+	}
+	return result
+}
+
+// ConvertFrom converts pdata.Logs into a slice of entry.Entry
+// To be used in a stateless setting like tests where ease of use is more
+// important than performance or throughput.
+func ConvertFrom(pLogs pdata.Logs) []*entry.Entry {
+	result := []*entry.Entry{}
+	for i := 0; i < pLogs.ResourceLogs().Len(); i++ {
+		rls := pLogs.ResourceLogs().At(i)
+		for j := 0; j < rls.ScopeLogs().Len(); j++ {
+			scope := rls.ScopeLogs().At(j)
+			result = append(result, convertFromLogs(fromConverterWorkerItem{Resource: rls.Resource(), LogRecordSlice: scope.LogRecords()})...)
+		}
+	}
+	return result
+}
+
+// convertFrom converts pdata.LogRecord into provided entry.Entry.
+func convertFrom(src pdata.LogRecord, ent *entry.Entry) {
+	ent.Timestamp = src.Timestamp().AsTime()
```

Comment:
A couple gotchas here. 
1. `pdata.Timestamp` is `uint64`, while `entry.Entry.Timestamp` is `time.Time`. These representations have different zero values, so we need to account for this.
2. We need to handle ObservedTimestamp here as well. Technically, it should have been initialized upstream, but if we don't check it could end up being nonsense.

I believe the correct conversion in this case looks like:
```suggestion
	 // if src.Timestamp == 0, then leave ent.Timestamp as nil
	if src.Timestamp != 0 {
		ent.Timestamp = src.Timestamp().AsTime()
	}
	
	if src.ObservedTimestamp == 0 {
		ent.ObservedTimestamp = time.Now()
	} else {			
		ent.ObservedTimestamp = src.ObservedTimestamp().AsTime()
	}
```